src/SFML/Window/Unix/WindowImplX11.cpp

```diff
@@ -981,7 +983,12 @@ void WindowImplX11::setSize(const Vector2u& size)
         XFree(sizeHints);
     }
 
-    XResizeWindow(m_display, m_window, size.x, size.y);
+    m_resizeOccuredDuringSetSize = false;
+    while(getSize() != size && !m_resizeOccuredDuringSetSize)
+    {
+        processEvents();
+        XResizeWindow(m_display, m_window, size.x, size.y);
+    }
```

Comment:
Spinning loop with X-calls is unlikely the way to go 🤔

Reply:
Hmmm. I guess I'll have to tinker with other ideas later. I tried setting the mouse position in the window.setSize function, but that unfortunately doesn't really solve the issue.

Reply:
I added a timeout to force quit this loop. Do another code review on the new code.